### PR TITLE
Python wrap enhanced stereo setters

### PR DESCRIPTION
**Pull request by d-b-w**
#### What does this implement/fix? Explain your changes.
This wraps creation of Enhanced Stereo groups from Python. It
also allows setting enhanced stereo groups on an RWMol from
Python.

This is required for an internal project where we are building
Mol objects in some Python code.

#### Any other comments?
Additionally, this provides a little function to allow C++
vectors to be wrapped for Python, but allow Python list objects
or wrapped C++ vectors to be used as arguments to functions
that take vectors as inputs. I added it only to
Wrap/StereoGroup.cpp, but I _think_ that I should add this
logic to RegisterVectorConverter in RDBoost/Wrap.h. If I
did that, I'd be able to remove a couple of lines in Wrap/Mol.cpp

### CODE REVIEW

File: Code/GraphMol/Wrap/rough_test.py [hunk 1]
```diff
@@ -5051,6 +5051,34 @@ def testEnhancedStereoPreservesMol(self):
...
+    groups = m2.GetStereoGroups()
+    self.assertEqual(len(groups), 1)
+    # Can set the StereoGroups using an empty list
+    m2.SetStereoGroups([])
+    self.assertEqual(len(m2.GetStereoGroups()), 0)
+
+    # Can create a StereoGroup using a Python list
+    group1 = Chem.rdchem.StereoGroup(Chem.rdchem.StereoGroupType.STEREO_OR, [m.GetAtomWithIdx(1)])
+    # Can set the StereoGroups using a Python list
+    m2.SetStereoGroups([group1])
```
Comment (greglandrum) [inline Code/GraphMol/Wrap/rough_test.py]: This highlights one of the things I'm worried about here: `group1` is defined using an atom from `m`, but we're able to set it as a stereogroup on `m2`. The result is that the following dumps core:
```
        m = Chem.MolFromSmiles('F[C@@H](Br)[C@H](F)Cl |o1:1|')
        m2 = Chem.RWMol(m)

        # Can create a StereoGroup using a Python list
        group1 = Chem.rdchem.StereoGroup(
            Chem.rdchem.StereoGroupType.STEREO_OR, [m.GetAtomWithIdx(1)])
        # Can set the StereoGroups using a Python list
        m2.SetStereoGroups([group1])
        self.assertEqual(len(m2.GetStereoGroups()), 1)

        m = None
        group1 = None
        gc.collect()

        sgs = m2.GetStereoGroups()
        self.assertEqual(len(sgs), 1)
        stereo_atoms = sgs[0].GetAtoms()
        self.assertEqual(stereo_atoms[0].GetIdx(), 1)
        self.assertEqual(stereo_atoms[0].GetOwningMol().GetNumAtoms(), 6)
```

Comment (greglandrum) [reply Code/GraphMol/Wrap/rough_test.py]: It's user error, but it really shouldn't be possible to trigger a core dump from Python

**Comment by greglandrum** [comment]
I made some proposed changes to (hopefully) fix the lifetime issues and did a PR against your branch

File: Code/GraphMol/Wrap/rough_test.py (hunk 1 above)
Comment (d-b-w) [reply Code/GraphMol/Wrap/rough_test.py]: yikes! Thanks.

**Comment by d-b-w** [comment]
We now have two ways to build StereoGroup objects from Python.

* `group = rdchem.createAndReturnStereoGroup(m, [1, 4], rdchem.STEREO_OR)`
* `group = rdchem.StereoGroup(rdchem.STEREO_OR, [atom1, atom4])`

It feels like we should remove one of these and make the other the default constructor. Do Python users usually use index-based identifiers to refer to atom objects? Or do you think that they get atoms from some sort of filtering function that returns `Atom*`?

**Comment by greglandrum** [comment]
That’s a good point. We are almost certainly fine with just using indices.

**Comment by greglandrum** [comment]
@d-b-w : do you want to remove the ctor that takes a list of atoms or shall I go ahead and merge this as-is?

**Comment by d-b-w** [comment]
Let me work a little more on removing the constructor. I'll post an update Monday.

**Comment by greglandrum** [comment]
Sounds good

**Comment by d-b-w** [comment]
Ugh, I just couldn't convince boost::python to let me use `with_custodian_and_ward_postcall()` with the constructor, so I just dropped the constructor. I think that users will survive with this factory function.

**Comment by greglandrum** [comment]
Thanks @d-b-w !

**Pull request merged**